### ISSUE TITLE
Яндекс.Практикум: Код-ревью.

### ISSUE THREAD

**Issue by zlocate** [SergeiOsokin/sergeiosokin.github.io#1]
Здравствуйте. Благодарю за проделанную работу.
К сожалению, в работе присутствуют недостатки не позволяющие ее засчитать на данной итерации. После их исправления работа будет засчитана (в случае если в процессе исправления не возникнет иных ошибок).

Репозиторий:

1. **Нужно исправить:** Отсутствуют обязательные ветки (несоблюдение ветвления): gh-pages, develop
2. **Нужно исправить:** Отсутствует версионирование при при помощи git-тегов (можно сделать так, например, `git tag -a v0.0.1 -m "my first version"`)
Подробнее о git-тегах: https://git-scm.com/book/ru/v2/%D0%9E%D1%81%D0%BD%D0%BE%D0%B2%D1%8B-Git-%D0%A0%D0%B0%D0%B1%D0%BE%D1%82%D0%B0-%D1%81-%D0%BC%D0%B5%D1%82%D0%BA%D0%B0%D0%BC%D0%B8

**Регулярные выражения.**

Имя: 
**Нужно исправить:** Проходит валидацию некорректное имя `Денис-` (неполное двойное)

Сайт:
1. **Нужно исправить:** Адрес `http://.ru` засчитывается как корректный. (некорректное (пустое) доменное имя)
2. **Нужно исправить:** Адрес с некорректным (неполным) IP адресом `http://8.8.8` засчитывается как корректный
3.  **Нужно исправить:** Адрес с некорректно заданным протоколом проходит валидацию. Пример: `http:ya.travel`

Номер телефона:
**Нужно исправить:** Не проходят валидацию следующие корректные номера телефонов:
1. `+7(925)900-90-90`
2. `+7(925) 900-90-90`
3. `+7 925-900-90-90`
4. `+79259009090`
5. `89259009090`

**Можно лучше:** добавить в репозиторий файл .gitignore, предотвращающий возможное попадение в репозиторий временных файлов

**Comment by zlocate** [comment]
**Исправление валидации поля имя засчитано**.

К сожалению не все проблемы ушли после исправлений.
**Нужно исправить**: расходятся версии проекта в разных ветках. Пожалуйста слейте ветки так чтобы во всех трех был одинаковый код.
**Нужно исправить**:  неполное README оформленное readme.md. Нужно:
проставленная версия, заголовок, ссылка на проект. 
Пожалуйста обратите внимание чтобы эти блоки не сливались в одну строку.

В ветке `master` одновременно указано 2 версии 1 и 2. Так быть не должно :(
В в ветке `develop` ссылки нет вовсе.

Если вам будет любопытно рассмотреть примеры хорошего оформления README
https://github.com/othneildrew/Best-README-Template
https://github.com/matiassingers/awesome-readme

**Нужно исправить:** Отсутствует версионирование при при помощи git-тегов (можно сделать так, например, `git tag -a v0.0.1 -m "my first version"`)
Подробнее о git-тегах: https://git-scm.com/book/ru/v2/%D0%9E%D1%81%D0%BD%D0%BE%D0%B2%D1%8B-Git-%D0%A0%D0%B0%D0%B1%D0%BE%D1%82%D0%B0-%D1%81-%D0%BC%D0%B5%D1%82%D0%BA%D0%B0%D0%BC%D0%B8
Есть понимание каким образом решить данную проблему? Нужна помощь?

**Нужно исправить**: Адрес http://.ru засчитывается как корректный. (некорректное (пустое) доменное имя)
Проблема осталась: https://prnt.sc/rdenf4
**Нужно исправить**: Адрес с некорректным (неполным) IP адресом `http://8.8.8` засчитывается как корректный
Проблема осталась. https://prnt.sc/rdenv5

**Нужно исправить**: Адрес с некорректно заданным протоколом проходит валидацию. Пример: `http:ya.travel`
Проблема осталось

После исправления регулярного выражения номера телефона некорректные номера воспринимаются как корректные.
Пример: `+7(925)900-90`, `+7 925-900-`, `79259009###`, `89#59Z090N0` и даже `NOPHONENUMBER` 
https://prnt.sc/rdeplh

**Comment by zlocate** [comment]
**Можно лучше:** в репозитории лежат файлы logo.svg и cover,jpg, однако фактически изображения загружаются из внешнего источника (серверов Яндекса)
Пожалуйста измените пути для локальной загрузки или удалите файлы из репозитория.

**Comment by SergeiOsokin** [comment]
Добрый день!

**Comment by SergeiOsokin** [comment]
Есть понимание каким образом решить данную проблему? Нужна помощь?
Да, если у Вас есть такая возможность...
После каждого коммита писал эти теги... И git tag тоже показывает, что всё есть.

**Comment by zlocate** [comment]
Обратите внимание что теги пушатся по-умолчанию отдельно.
Вы команду ` git push origin --tags` выполняли?

Подробнее тут: https://git-scm.com/book/ru/v2/%D0%9E%D1%81%D0%BD%D0%BE%D0%B2%D1%8B-Git-%D0%A0%D0%B0%D0%B1%D0%BE%D1%82%D0%B0-%D1%81-%D0%BC%D0%B5%D1%82%D0%BA%D0%B0%D0%BC%D0%B8

**Comment by SergeiOsokin** [comment]
Да, я уже увидел. Вроде выложил... Подскажите, пожалуйста, когда я работаю в develop, 
правильно ведь такая последовательность: git add,  потом commit,  затем tag, потом push, потом push tags и только после выполняют merge например в master, нужно ли потом в master выполнять какие либо push? Или слияние все "само" делает, не могу уловить.

**Comment by zlocate** [comment]
Добрый вечер :)
Теги прилетели в репозиторий, все правильно! 
В целом последовательность действий, которая у вас указана верная.
После merge необходимо сделать push.

Обращаю ваше внимание, что в VSCode есть средства, позволяющие все это сделать визуально,
На всякий случай приведу скриншоты для описания основных действий.
а) Добавить файл к отслеживанию (аналог git add)
https://prnt.sc/rdgp5w (переключиться на вкладку изображающей ветку системы контроля версии (три точки), нажать плюс рядом с именем файла.
б) Создать коммит.
Ввести текст в поле ввода, нажать на галку над полем (commit)
https://prnt.sc/rdgpzd
в) Отправить коммит в репозиторий (push) - зеленая стрелка
г) Переключить ветку (красная стрелка)
https://prnt.sc/rdgqzh
Получается удобно и достаточно быстро, особенно на первых этапах, когда команды git'а не совсем уложились в голове :)

**Comment by zlocate** [comment]
Если будут еще вопросы - пишите :)
А так жду правок по остальным комментариям.

**Comment by zlocate** [comment]
Добрый день. **С последней проверки заметен значительный прогресс.**
Проблемы, описанные выше решены, новые - не выявлены.
**Работа засчитана!**
Желаю вам успехов в дальнейшем изучении курса. Еще увидимся!

**Comment by SergeiOsokin** [comment]
Спасибо, большое!)
